lib/ringbuffer/ring_buffer_backend.c

```diff
@@ -12,11 +12,11 @@
 #include <linux/delay.h>
```

Comment:
about the changelog: the 

{set, clear}_current_oom_origin() functions are trivially wrapped with an
empty definition when they are unavailable.

does not match the implementation, which adds wrappers for those (the implementation is OK, the changelog is wrong).

Reply:
What I meant to say is that the functions are trivially wrapped when the function are available AND defined empty when they are unavailable. I will improve it.

Reply:
I changed it to the following: 

> wrapper_check_enough_free_pages() uses the si_mem_available() kernel
function to compute if the number pages requested passed as parameter is
smaller than the number of pages available on the machine. If the
si_mem_available() kernel function is unavailable, we always return
true.
> wrapper_set_current_oom_origin() function wraps the
set_current_oom_origin() kernel function when it is available.
If set_current_oom_origin() is unavailable the wrapper is empty.
> wrapper_clear_current_oom_origin() function wraps the
clear_current_oom_origin() kernel function when it is available.
If clear_current_oom_origin() is unavailable the wrapper is empty.

What do you think?